elasticdl/doc/async_sgd_design.md

```diff
@@ -0,0 +1,115 @@
+# Design Doc: Asynchronous SGD
+
+## Motivation
+Parameter server (PS) based distributed training uses data parallelism to speed up training. 
+
+We have implemented synchronous SGD in ElasticDL. When PS accumulates `grads_to_wait` gradients from workers, PS averages these gradients and updates the model with the averaged gradients. PS also maintains `model_version` which equals to the number of model updates. Each worker has a local copy of the model. Before a minibatch training step, if there is a new `model_version` on PS, the worker will get the new model from PS to replace the local model. After computing the gradients with a minibatch, the worker reports the gradients to PS together with the local model version. When PS receives gradients from a worker, it only accepts the gradients with a model version same as the current PS `model_version`.
```

Comment:
`Before a minibatch training step` is not a complete clause? Change it to `Before a minibatch training step starts`?